test-runner/adapters/direct_python_sdk/direct_module_api.py

```diff
@@ -8,48 +8,49 @@
 import json
 from ..print_message import print_message
 from ..abstract_module_api import AbstractModuleApi
-from azure.iot.hub.devicesdk import ModuleClient
+from azure.iot.hub.devicesdk.sync_clients import ModuleClient as ModuleClientSync
 from azure.iot.hub.devicesdk.auth.authentication_provider_factory import (
     from_connection_string,
     from_environment,
 )
 
 # logging.basicConfig(level=logging.INFO)
 
-object_list = []
+object_list_sync = []
```

Comment:
Consider making this an attribute of the class instead of a global.

NOT an attribute of a class instance, but of the class itself, shared between all instances, and available even there are no class instances via ModuleApi.object_list_sync.

This is not only cleaner, by keeping it self contained, but also reduces your imports when it gets used in the __init__